import.go

```diff
@@ -59,6 +60,13 @@ func WithDigestRef(f func(digest.Digest) string) ImportOpt {
 	}
 }
 
+// WithoutDuplicatedDigestRef is used to prevent a manifest to have a
+// digest image if the image is already created.
+func WithoutDuplicatedDigestRef(c *importOpts) error {
```

Comment:
Needs a definition of “digest image”

Reply:
Can we have a flag for this in ctr CLI?

Reply:
> Needs the definition of “digest image”

Fixed the comment to make it clearer.

> Can we have a flag for this in ctr CLI?

Added `--dedupe` option.

Reply:
The options are still a little confusing since this isn't really duplication of any specific type.

I have wanted to move the image references toward a model of `<tag>` and `<tag>@<digest>`, and have the "untagged" `@<digest>` treated by the client as untagged images. Then you could also see the different versions of the tags and it is up to the client to collapse the display and delete all associated tags when an image is removed. I agree that just the digest reference here makes no sense. Maybe the functionality though is "skip-digest-only" references for when there is a name for now.

Reply:
for me the opposite makes more sense for an end user; omit the digest tag by default (if the name tag exists of course) and keep the digest if the `--digest` tag is present. I don't see the use of a digest tag if a user has not recommended it

Reply:
Fixed to use the term "skip" instead of "dedupe".

Reply:
@fahedouch The "end user behavior" (referring to a typical developer using nerdctl) can and should be different from the behavior of `ctr` and the containerd client library. `ctr` is focused on giving operators access to the various APIs and client library functionality through the command line with the expectation that those operators understand how containerd works. The client library changes should be focused on flexibility to support any functionality a client may have. `nerdctl` on the other hand should focus on end user expectations and making sure that it is usable with little knowledge of containerd. That is at least my perspective when proposing or rejecting changes related to `ctr`.